R/pipelines.R

```diff
@@ -519,9 +519,68 @@ pipeline_survtoregr = function(method = 1, regr_learner = lrn("regr.featureless"
   gr
 }
 
+#' @name mlr_graphs_survtoclassif
+#' @title Survival to Classification Reduction Pipeline
+#' @description Wrapper around multiple [PipeOp][mlr3pipelines::PipeOp]s to help in creation
+#' of complex survival reduction methods.
+#'
+#' @param learner [LearnerClassif][mlr3::LearnerClassif]\cr
+#' Classification learner to fit the transformed [TaskClassif][mlr3::TaskClassif].
+#' `learner` must have `predict_type` of type `"prob"`.
+#' @param cut `numeric()`\cr
+#' Split points, used to partition the data into intervals.
+#' If unspecified, all unique event times will be used.
+#' @param max_time `numeric(1)`\cr
+#' If cut is unspecified, this will be the last possible event time.
+#' All event times after max_time will be administratively censored at max_time.
+#' @param rhs `character(1)`\cr
+#' Right-hand side of the formula to with the learner.
+#' All features of the task are available as well as `tend` the upper bounds
+#' of the intervals created by `cut`.
+#' If rhs is unspecified, the formula of the task will be used.
+#'
+#' @return [mlr3pipelines::Graph] or [mlr3pipelines::GraphLearner]
+#' @family pipelines
+#'
+#' @examples
+#' \dontrun{
+#' if (requireNamespace("mlr3pipelines", quietly = TRUE)) {
+#'   library("mlr3")
+#'   library("mlr3pipelines")
+#'
+#'   task = tsk("rats")
+#'
+#'   pipe = ppl(
+#'     "survtoclassif",
+#'     learner = lrn("classif.log_reg")
+#'   )
+#'   pipe$train(task)
+#'   pipe$predict(task)
+#'
+#' }
+#' }
+#' @export
+pipeline_survtoclassif = function(learner, cut = NULL, max_time = NULL, rhs = NULL) {
+  if (!is.null(rhs)) {
+    gr = mlr3pipelines::po("trafotask_survclassif", cut = cut, max_time = max_time) |>
```

Comment:
compress the code, see other pipelines how to incrementally add the pipeops and in step 2 if rhs is null, don't include the intermediate pipeop